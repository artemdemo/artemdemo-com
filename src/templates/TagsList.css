.tag-link {
  display: inline-block;
  margin-right: 15px;
}
