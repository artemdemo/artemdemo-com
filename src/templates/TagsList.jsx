import React from 'react';
import _get from 'lodash/get';
import { graphql, Link } from 'gatsby';
import styled from 'styled-components';
import Layout from '../components/Layout/Layout';

const TagLinkSty = styled(Link)`
    display: inline-block;
    margin-right: 15px;
`;

class TagsList extends React.PureComponent {
    constructor(props) {
        super(props);
    }

    renderTags() {
        const tagsMap = _get(this.props, 'pageContext.tagsMap');
        return Object.keys(tagsMap)
            .map(slug => ({
                slug,
                name: tagsMap[slug].name,
                amount: tagsMap[slug].amount,
            }))
            .sort((tagA, tagB) => tagB.amount - tagA.amount)
            .map(tag => (
                <TagLinkSty
                    to={`/tags/${tag.slug}`}
                    key={`tags-${tag.slug}`}
                >
                    {tag.name}
                    <sup> {tag.amount}</sup>
                </TagLinkSty>
            ));
    }

    render() {
        const siteTitle = _get(this.props, 'data.site.siteMetadata.title');
        const siteDescription = _get(this.props, 'data.site.siteMetadata.description');
        const title = 'Tags List';

        return (
            <Layout
                title={`${title} | ${siteTitle}`}
                description={siteDescription}
                metaTitle={siteTitle}
                location={this.props.location}
            >
                <h1>{title}</h1>

                {this.renderTags()}
            </Layout>
        );
    }
}

export default TagsList;

export const pageQuery = graphql`
  query {
    site {
      siteMetadata {
        title
        description
      }
    }
  }
`;
