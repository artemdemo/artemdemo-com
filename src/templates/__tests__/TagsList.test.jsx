import React from 'react';
import renderer from 'react-test-renderer';
import TagsList from '../TagsList';

jest.mock('gatsby');
jest.mock('../../components/Layout/Layout');

describe('TagsList', () => {
    it('should render empty', () => {
        const tree = renderer.create(
            <TagsList
                data={{}}
                pageContext={{}}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with data', () => {
        const tree = renderer.create(
            <TagsList
                data={{
                    site: {
                        siteMetadata: {
                            title: 'Test title',
                            description: 'Some description',
                        },
                    }
                }}
                pageContext={{
                    tagsMap: {
                        tag1: {
                            name: 'tag1',
                            amount: 10,
                        },
                        tag2: {
                            name: 'tag2',
                            amount: 10,
                        },
                    },
                }}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});