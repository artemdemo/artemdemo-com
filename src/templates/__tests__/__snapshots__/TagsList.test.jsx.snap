// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsList should render empty 1`] = `
<div
  data-mock="Layout"
  data-title="Tags List | undefined"
>
  <h1>
    Tags List
  </h1>
</div>
`;

exports[`TagsList should render with data 1`] = `
<div
  data-description="Some description"
  data-metaTitle="Test title"
  data-mock="Layout"
  data-title="Tags List | Test title"
>
  <h1>
    Tags List
  </h1>
  <div
    data-mock="Link"
  >
    {
  "to": "/tags/tag1",
  "className": "TagsList__TagLink-sc-2p4wkd-0 jInolM"
}
    tag1
    <sup>
       
      10
    </sup>
  </div>
  <div
    data-mock="Link"
  >
    {
  "to": "/tags/tag2",
  "className": "TagsList__TagLink-sc-2p4wkd-0 jInolM"
}
    tag2
    <sup>
       
      10
    </sup>
  </div>
</div>
`;
